apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloExtension.groovy

```diff
@@ -0,0 +1,34 @@
+package com.apollostack.android.gradle
+
+import com.google.common.collect.ArrayListMultimap
+import com.google.common.collect.Multimap
+import org.gradle.api.Project
+import org.gradle.api.tasks.util.PatternSet
+
+class ApolloExtension {
+  private final Project project
+  static String NAME = "apollo"
+  String graphqlPath
+  // Maps GraphQL query files' relative paths to a sourceSet directory
+  private Multimap<String, String> cachedFiles
+
+  ApolloExtension(Project project, String sourceSet) {
+    this.project = project
+    this.graphqlPath = "src/$sourceSet/graphql"
+  }
+
+  Multimap<String, String> getFiles() {
+    if (cachedFiles != null) {
+      return cachedFiles
+    }
+    PatternSet patternSet = new PatternSet().include("**/*.graphql")
+    ArrayListMultimap<String, String> files = ArrayListMultimap.create()
+    project.files(graphqlPath).getAsFileTree().matching(patternSet).visit { el ->
+      if (!el.directory) {
```

Comment:
I need the call to `visit` because `relativePath` exists only on a `FileVisitDetails` object. A filter would imply a `File` type.